src/Product/AttributeMapping/AttributeMappingHelper.php

```diff
@@ -0,0 +1,316 @@
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Helper Class for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product\AttributeMapping
+ */
+class AttributeMappingHelper implements Service {
+
+
+	private const ATTRIBUTES_AVAILABLE_FOR_MAPPING = [
+		Adult::class,
+		AgeGroup::class,
+		Brand::class,
+		Color::class,
+		Condition::class,
+		Gender::class,
+		GTIN::class,
+		IsBundle::class,
+		Material::class,
+		MPN::class,
+		Multipack::class,
+		Pattern::class,
+		Size::class,
+		SizeSystem::class,
+		SizeType::class,
+	];
+
+	public const CATEGORY_CONDITION_TYPE_ALL    = 'ALL';
+	public const CATEGORY_CONDITION_TYPE_ONLY   = 'ONLY';
+	public const CATEGORY_CONDITION_TYPE_EXCEPT = 'EXCEPT';
+
+	/**
+	 * @var Transients|null
+	 */
+	private static ?Transients $transients = null;
+
+	/**
+	 * Gets all the available attributes for mapping
+	 *
+	 * @return array
+	 */
+	public function get_attributes(): array {
+		$destinations = [];
+
+		/**
+		 * @var WithMappingInterface $attribute
+		 */
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			array_push(
+				$destinations,
+				[
+					'id'    => $attribute::get_id(),
+					'label' => $attribute::get_name(),
+					'enum'  => $attribute::is_enum(),
+				]
+			);
+		}
+
+		return $destinations;
+	}
+
+	/**
+	 * Gets all the available sources identified by attribute key
+	 *
+	 * @return array
+	 */
+	public function get_sources(): array {
+		$sources = [];
+
+		/**
+		 * @var AttributeInterface $attribute
+		 */
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+
+			$attribute_sources = [];
+
+			foreach ( $attribute::get_sources() as $key => $value ) {
+				array_push(
+					$attribute_sources,
+					[
+						'id'    => $key,
+						'label' => $value,
+					]
+				);
+			}
+
+			$sources[ $attribute::get_id() ] = $attribute_sources;
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * Returns a single instance of Transients
+	 *
+	 * @return Transients A single instance of Transients
+	 */
+	public static function get_transients_object(): Transients {
+		if ( is_null( self::$transients ) ) {
+			self::$transients = new Transients();
+		}
+
+		return self::$transients;
+	}
+	/**
+	 * Gets the taxonomies and global attributes to render them as options in the frontend.
+	 *
+	 * @return array An array with the taxonomies and global attributes
+	 */
+	public static function get_source_taxonomies(): array {
+		$taxonomies        = get_object_taxonomies( 'product', 'objects' );
+		$parsed_taxonomies = [];
+		$attributes        = [];
+		$sources           = [];
+
+		foreach ( $taxonomies as $taxonomy ) {
+			if ( taxonomy_is_product_attribute( $taxonomy->name ) ) {
+				$attributes[ 'taxonomy:' . $taxonomy->name ] = $taxonomy->label;
+				continue;
+			}
+
+			$parsed_taxonomies[ 'taxonomy:' . $taxonomy->name ] = $taxonomy->label;
+		}
+
+		asort( $parsed_taxonomies );
+		asort( $attributes );
```

Comment:
Yes. my initial idea was to show the taxonomies/attributes added by the hook after everything else. Relying on the client to decide how it should be sorted. But also fine for me to force our sorting after.

Reply:
It's fine to leave as is, was just wondering if there was a specific need to have them sorted. Most extensions that will hook into this will probably just add without sorting.